storage/objects.go

```diff
@@ -304,3 +321,48 @@ func (s *ObjectsClient) Put(ctx context.Context, input *PutObjectInput) error {
 
 	return nil
 }
+
+func createDirectory(c ObjectsClient, ctx context.Context, fullPath string) error {
+	log.Printf("Forcing the Creation of the Directory Tree: %q", fullPath)
+	dirClient := &DirectoryClient{
+		client: c.client,
+	}
+
+	parts := strings.Split(fullPath, "/")
+	parts = parts[:len(parts)-1]
+
+	var folderPath string
+	for i, part := range parts[1:] {
+		if i == 0 {
+			folderPath = part
+			continue
+		}
+		folderPath = path.Clean(path.Join(folderPath, part))
+		err := dirClient.Put(ctx, &PutDirectoryInput{
+			DirectoryName: folderPath,
+		})
+		if err != nil {
+			return err
```

Comment:
Will this error out if we can't create a directory along the way for whatever reason?

Reply:
yes, it will error out and report that error to the user